### PR TITLE
[FIX] stock: allow to delete an progress inventory ajustement

### PR DESCRIPTION
**Pull request by fmdl**
Before that it was possible to delete a started inventory (sale, purchase, invoice it is not possible)

@jco-odoo

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by jco-odoo** [comment]
Why could not you delete a confirmed inventory adjustment?

**Comment by fmdl** [comment]
For sale, invoice, purchase you can not delete an uncancelled record, you need to cancel before:
https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L185
https://github.com/odoo/odoo/blob/11.0/addons/account/models/account_invoice.py#L606
https://github.com/odoo/odoo/blob/11.0/addons/purchase/models/purchase.py#L205

It can be confusing for the user. It is not a big issue, we can fix that in master.

File: addons/stock/models/stock_inventory.py [hunk 1]
```diff
@@ -99,8 +99,8 @@ def _compute_total_qty(self):
     @api.multi
     def unlink(self):
         for inventory in self:
-            if inventory.state == 'done':
-                raise UserError(_('You cannot delete a validated inventory adjustement.'))
+            if inventory.state not in ('draft', 'cancel'):
+                raise UserError(_('You cannot delete an inventory adjustement which is not draft or cancelled.'))
```
Comment (mart-e) [inline addons/stock/models/stock_inventory.py]: that breaks the existing translations, see https://github.com/odoo/odoo/wiki/Contributing#how-should-i-handle-translatable-content

**Comment by fmdl** [comment]
@mart-e my PR it is correct now ?

**Comment by mart-e** [comment]
@fmdl on the translation part yes but I let @jco-odoo decide on the purpose

**Comment by fmdl** [comment]
ping @jco-odoo

**Comment by fmdl** [comment]
@simongoffin

**Comment by fmdl** [comment]
ping @jco-odoo

**Comment by fmdl** [comment]
ping @jco-odoo

**Comment by jco-odoo** [comment]
@Whenrow @amoyaux He has a (minor) point.  As it is for master, why not?

**Comment by amoyaux** [comment]
@fmdl Ok I will modify the commit message and push it

**Comment by fmdl** [comment]
@jco-odoo any news ?

**Comment by jco-odoo** [comment]
@amoyaux has not this been done already by any chance?

**Comment by fmdl** [comment]
@jad-odoo and @amoyaux it is not already done in master https://github.com/odoo/odoo/blob/master/addons/stock/models/stock_inventory.py#L104

**Comment by fmdl** [comment]
ping @jco-odoo

**Comment by jco-odoo** [comment]
@sle-odoo @amoyaux @lap-odoo

**Comment by lap-odoo** [comment]
@fmdl That's ok for me but there are two things to change 
- The error message, can you replace it by : 'You can only delete a draft inventory adjustment. If the inventory adjustment is not done, you can cancel it.'
- And review commit message : [REF] stock: unlink of inventory adjustment
allow deletion of draft and cancelled inventory adjustment 

Thanks 

@

**Comment by fmdl** [comment]
@lap-odoo it is changed. thanks

**Comment by lap-odoo** [comment]
@fmdl Thanks. @sle-odoo Can you please check it and merge ? Thanks

**Comment by sle-odoo** [comment]
thanks!
robodoo r+

**Comment by sle-odoo** [comment]
rebased in #34469

**Pull request closed without merging**